packages/react-server/src/ReactFizzServer.js

```diff
@@ -483,6 +484,48 @@ function renderSuspenseBoundary(
   // context switching. We just need to temporarily switch which boundary and which segment
   // we're writing to. If something suspends, it'll spawn new suspended task with that context.
   task.blockedBoundary = newBoundary;
```

Comment:
This is saying that this task is blocking the completion of the new boundary, but the fallback isn't actually blocking this boundary. It's blocking the parent boundary from completing since it can't show anything without the fallback. You should probably add a test for suspending inside the fallback and that delaying the parent.

Note how `suspendedFallbackTask = createTask` below passes the parentBoundary as the blockedBoundary for the fallback.

In other words, this line should be after the `if` statement.